### PR TITLE
fixed compiler error message on wildcard pattern within expression

### PR DESCRIPTION
**Pull request by 00AR**
Signed-off-by: Abdul Rafey <abdulrafeyq@gmail.com>

fixed compiler error message on wildcard pattern '_'  within expression.
Added a new testcase for the same.
#867

### CODE REVIEW

File: gcc/rust/parse/rust-parse-impl.h [hunk 1]
```diff
@@ -12900,6 +12900,12 @@ Parser<ManagedTokenSource>::null_denotation (const_TokenPtr tok,
     case UNSAFE:
       return parse_unsafe_block_expr (std::move (outer_attrs),
 				      tok->get_locus ());
+    case UNDERSCORE:
+      add_error (
+	Error (tok->get_locus (),
+	       "Use of %qs is not allowed on the right-side of an assignment",
```
Comment (CohenArthur) [inline gcc/rust/parse/rust-parse-impl.h:12906]:
```suggestion
	       "use of %qs is not allowed on the right-side of an assignment",
```

We usually use lowercase error messages, as it looks better with GCC's internal formatter

**Review by CohenArthur — APPROVED** [review]
Looks good to me apart from the very tiny nitpick :) Once you fix it ping me and I'll merge this PR. Thanks for working on this!

File: gcc/rust/parse/rust-parse-impl.h (hunk 1 above)
Comment (00AR) [reply gcc/rust/parse/rust-parse-impl.h]: I have made the change. @CohenArthur Thanks for reviewing!

File: gcc/testsuite/rust/compile/issue-867.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+fn main() {
+    let _ = 42;
+    let a = _ + 123; // { dg-error "Use of '_' is not allowed on the right-side of an assignment" }
```
Comment (CohenArthur) [inline gcc/testsuite/rust/compile/issue-867.rs:3]: This test is going to fail now :DD but no worries it's not a big deal. Once that's fixed I'll merge it :)

**Review by CohenArthur — APPROVED** [review]
Last stretch haha. Sorry! I should have thought about it in my previous review

File: gcc/testsuite/rust/compile/issue-867.rs (hunk 2 above)
Comment (00AR) [reply gcc/testsuite/rust/compile/issue-867.rs]: > This test is going to fail now

became a bit lazy :D 
Fixed the testcase and this time made sure it passes. Also I rebased this PR too.

**Review by CohenArthur — APPROVED** [review]
Great stuff :) Thanks for taking the time!

bors r+

**Pull request merged**